lib/gemini.js

```diff
@@ -23,39 +26,49 @@ function requireWithNoCache(moduleName) {
     return result;
 }
 
-function Gemini(config, overrides) {
-    config = config || DEFAULT_CFG_NAME;
-    this.config = new Config(config, overrides);
-    var _this = this;
+module.exports = inherit(EventEmitter, {
+    __constructor: function(config, overrides) {
+      config = config || DEFAULT_CFG_NAME;
```

Comment:
Yes I share your concerns - a problem indeed arises with sauce-connect because it needs to finish before running the tests should begin. I thought about it, but originally couldn't find a suitable place in gemini to hook a callback or a promise to it. 
So, instead I was planning just to block the execution in the sauce plugin until it's ready - and yes, I'd really rather do it properly :-)

I think I'll start by looking for options on how to do it using promises.